sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -411,7 +411,16 @@ public void updateAddresses(final URI serverKey) {
             if (!(exception instanceof CosmosException)) {
                 // wrap in CosmosException
                 logger.error("Network failure", exception);
-                dce = BridgeInternal.createCosmosException(request.requestContext.resourcePhysicalAddress, 0, exception);
+                int statusCode = 0;
```

Comment:
We have these if checks below , can we please use those to update the status code ?

Reply:
Also please add same logic in master partition path in the same file, we have duplicate code for server and master

Reply:
Good catch - fixed.